.circleci/getChangelogs.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+REQUIREMENTS_FILE="codacy/requirements.yaml"
+OLD_LOCK_FILE="codacy/requirements_old.lock"
+NEW_LOCK_FILE="codacy/requirements.lock"
+DEPENDENCIES=$(yq r $OLD_LOCK_FILE dependencies -j | jq ".[].name" | sed "s/\"//g")
+
+function getChangelog() {
+    CHANGELOG_FILENAME=$3
+    JIRA_CHANGELOG_FILENAME=$4
+    git fetch --all
+    git checkout tags/$1 --quiet
+    git changelog -s $1 -f $2 --stdout > ../changelogs/$CHANGELOG_FILENAME
+    git log $1..$2 --quiet | grep -iEow 'ft-[0-9]+' | sort | uniq > ../changelogs/$JIRA_CHANGELOG_FILENAME
+}
+
+function cloneRepository() {
+    PROJECT_NAME=$1
+    REPOSITORY_URL=$(yq r $REQUIREMENTS_FILE dependencies -j | jq ".[] | select(.name==\"$PROJECT_NAME\").git" | sed "s/\"//g")
+    OLD_VERSION=$2
+    NEW_VERSION=$3
+    CHANGELOG_FILENAME="$PROJECT_NAME-changelog.txt"
+    JIRA_CHANGELOG_FILENAME="$PROJECT_NAME-jira.txt"
+
+    
+    if [ "$REPOSITORY_URL" != "null" ];
+        then
+            echo "[OK] $PROJECT_NAME($REPOSITORY_URL) -> $OLD_VERSION : $NEW_VERSION"
+            git clone $REPOSITORY_URL $PROJECT_NAME --quiet
+            cd $PROJECT_NAME
+            getChangelog $OLD_VERSION $NEW_VERSION $CHANGELOG_FILENAME $JIRA_CHANGELOG_FILENAME;
+            cd ..
+            rm -rf ./$PROJECT_NAME
+        else
+            echo "[Skipped] $PROJECT_NAME has no repository url."
+    fi
+    
+}
+rm -rf ./changelogs
+mkdir changelogs
+for key in $DEPENDENCIES
+do
+    OLD_VERSION=$(yq r $OLD_LOCK_FILE dependencies -j | jq ".[] | select(.name==\"$key\").version" | sed "s/\"//g")
+    NEW_VERSION=$(yq r $NEW_LOCK_FILE dependencies -j | jq ".[] | select(.name==\"$key\").version" | sed "s/\"//g")
+    [ "$OLD_VERSION" != "$NEW_VERSION" ] && cloneRepository $key $OLD_VERSION $NEW_VERSION 
+done
+
+find ./changelogs -size 0 -delete # delete empty changelogs
```

Comment:
the -empty option doesn't work on alipine linux. I think it must be the gnu tools that are installed are older than the ones we have on our machines.